CHANGELOG.md

```diff
@@ -1,5 +1,96 @@
 # BEM-XJST Changelog
 
+# 2016-01-29, [v5.0.0](https://github.com/bem/bem-xjst/compare/v4.3.3...v5.0.0), @miripiruni
+
+Breaking changes: BEM-XJST now supports two template engines: BEMHTML for getting HTML output and BEMTREE for getting BEMJSON. By default BEM-XJST will use BEMHTML engine.
+Usage example:
+
+```js
+var bemxjst = require('bem-xjst');
+var bemhtml = bemxjst.bemhtml;
+
+bemhtml
+    // Add templates
+    .compile(function() {
+        block('b').content()('yay');
+    })
+    // Get result in HTML format
+    .apply(
+        // Data context in BEMJSON format
+        { block: 'b' }
+    );
+    // Result: <div class="b">yay</div>
+```
+
+```js
+var bemxjst = require('bem-xjst');
+var bemtree = bemxjst.bemtree;
+
+bemtree
+    // Add templates
+    .compile(function() {
+        block('b').content()('yay');
+    })
+    // Get result in BEMJSON format
+    .apply(
+        // Data context in BEMJSON format
+        { block: 'b' }
+    );
+    // Result: { block: 'b1', content: 'yay' }
+```
+
+
+Now support changing elemMods in runtime. Example:
+```js
+// Template
+block('b1').elem('e1').def()(function() {
+  this.elemMods.a = 'b';
+  return applyNext();
+});
+// BEMJSON
+{ block: 'b1', elem: 'e1' }
+// Result:
+'<div class="b1__e1 b1__e1_a_b"</div>'
+```
+
+BEM-XJST now should not treat mods as elemMods even if block exist.
+```
+// BEMJSON
+{
+  block: 'b',
+  elem: 'e',
+  mods: { m: 'v' } // will be ignored because of elem
+}
+// Result
+'<div class="b1__e1"></div>'
+```
+And should not treat elemMods as mods. Example:
+```js
+// BEMJSON
+{
+  block: 'b1',
+  elemMods: { m1: 'v1' }
+}
+// Result
+'<div class="b1"></div>'
+```
+
+BEMTREE will return true as is. Example:
```

Comment:
any Boolean actually (falsy input was handled in the first commit with BEMTREE implementation)